app/services/attachment_parser_service.rb

```diff
@@ -1,17 +1,30 @@
 class AttachmentParserService
-  def initialize(attachments:)
+  def initialize(attachments:, v2submission: false)
     @attachments = attachments
+    @v2submission = v2submission
   end
 
   def execute
     attachments.map do |attachment|
-      Attachment.new(
-        url: attachment.fetch(:url, nil),
-        mimetype: attachment.fetch(:mimetype),
-        filename: attachment.fetch(:filename),
-        type: attachment.fetch(:type),
-        path: nil
-      )
+      if @v2submission
```

Comment:
this does the same thing regardless of the value of v2submission?

Reply:
Yes it is either fetching a symbol :url or a string 'url'. It fails to parse the attachments if it is being given a symbol instead of a string so we have both cases. We could convert symbol hash to json and then parse json to string but the aim being to send a json to an api, I thought I would do that this way. I don't have other ideas.